lib/src/features/user/model/user.dart

```diff
@@ -134,3 +134,179 @@ class UserStatus with _$UserStatus {
+        deviation: perf('glicko', 'deviation').asDoubleOrThrow(),
+        provisional: perf('glicko', 'provisional').asBoolOrNull(),
+        progress: perf('progress').asIntOrThrow(),
+        rank: pick('rank').asIntOrNull(),
+        percentile: pick('percentile').asDoubleOrNull(),
+        totalGames: count('all').asIntOrThrow(),
+        berserkGames: count('berserk').asIntOrThrow(),
+        tournamentGames: count('tour').asIntOrThrow(),
+        ratedGames: count('rated').asIntOrThrow(),
+        wonGames: count('win').asIntOrThrow(),
+        lostGames: count('loss').asIntOrThrow(),
+        drawnGames: count('draw').asIntOrThrow(),
+        disconnections: count('disconnects').asIntOrThrow(),
+        avgOpponent: count('opAvg').asDoubleOrNull(),
+        timePlayed: count('seconds').asDurationFromSecondsOrThrow(),
+        lowestRating: lowest('int').asIntOrNull(),
+        lowestRatingGame: lowest.letOrNull(UserPerfGame.fromPick),
+        highestRating: highest('int').asIntOrNull(),
+        highestRatingGame: highest.letOrNull(UserPerfGame.fromPick),
+        curWinStreak: resultStreak('win', 'cur').letOrNull(UserStreak.fromPick),
+        maxWinStreak: resultStreak('win', 'max').letOrNull(UserStreak.fromPick),
+        curLossStreak:
+            resultStreak('loss', 'cur').letOrNull(UserStreak.fromPick),
+        maxLossStreak:
+            resultStreak('loss', 'max').letOrNull(UserStreak.fromPick),
+        curPlayStreak: playStreak('nb', 'cur').letOrNull(UserStreak.fromPick),
+        maxPlayStreak: playStreak('nb', 'max').letOrNull(UserStreak.fromPick),
+        curTimeStreak: playStreak('time', 'cur').letOrNull(UserStreak.fromPick),
+        maxTimeStreak: playStreak('time', 'max').letOrNull(UserStreak.fromPick),
+        worstLosses: stat('worstLosses', 'results')
+            .asListOrNull((pick) => UserPerfGame.fromPick(pick)),
+        bestWins: stat('bestWins', 'results')
+            .asListOrNull((pick) => UserPerfGame.fromPick(pick)));
+  }
+}
+
+@freezed
+class UserStreak with _$UserStreak {
+  const factory UserStreak.gameStreak({
+    required int gamesPlayed,
+    required bool isValueEmpty,
+    required UserPerfGame? startGame,
+    required UserPerfGame? endGame,
+  }) = UserGameStreak;
+
+  const factory UserStreak.timeStreak({
+    required Duration timePlayed,
+    required bool isValueEmpty,
+    required UserPerfGame? startGame,
+    required UserPerfGame? endGame,
+  }) = UserTimeStreak;
+
+  factory UserStreak.fromJson(Map<String, dynamic> json) {
+    return UserStreak.fromPick(pick(json).required());
+  }
+
+  factory UserStreak.fromPick(RequiredPick pick) {
+    final path = pick.path;
+    if (path.length <= 1) {
+      throw ArgumentError('pick must have a path longer than one');
+    }
+
+    // Since we are passing 'cur' or 'max' as the last pick,
+    // we check the previous value in the path (the parent).
+    final type = path.reversed.elementAt(1);
```

Comment:
Arguably this "problem" could be solved by creating a class that stores both the current streak and the longest streak, but I thought it would be better to leave it this way in order not to make these separated (although related) streaks coupled.
Even so, I can change this if you want to.